chirho/explainable/handlers/explanation.py

```diff
@@ -131,12 +148,43 @@ def SearchForExplanation(
         for a in antecedents.keys()
     }
 
+    if num_samples is not None:
+        if sampling_dim is None:
+            raise ValueError("sampling_dim must be provided if num_samples is provided")
+
+        case_shape = [1] * torch.abs(torch.tensor(sampling_dim))
+        case_shape[sampling_dim] = num_samples
+
+        antecedent_probs = torch.tensor(
+            [0.5 - antecedent_bias] + ([0.5 + antecedent_bias])
+        )
+
+        antecedent_case_dist = dist.Categorical(probs=antecedent_probs)
+
+        antecedent_cases = {
+            key: antecedent_case_dist.sample(case_shape) for key in antecedents.keys()
```

Comment:
A Pyro program should never be calling `Distribution.sample` directly and should never introduce batch dimensions that do not correspond to a `plate` context, otherwise the program's meaning and the correctness of any inference algorithm applied to it are undefined.